.github/workflows/pip-compile.yml

```diff
@@ -0,0 +1,77 @@
+---
+name: "Refresh pinned dependencies"
+
+"on":
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+  push:
+    branches:
+      - devel
+    paths:
+      - .github/workflows/pip-compile.yml
+      - "tests/*.in"
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  refresh:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: "devel"
+      - name: Graft ansible-core
```

Comment:
```suggestion
      - name: Fetch required contents of ansible-core
```

Reply:
hope it isn't too pedantic and I know "graft" is used in other workflows but I generally like really descriptive step names. feel free to take it or leave it.